### PR TITLE
provide ratelimit service/ deploy, also include related redis

### PR DESCRIPTION
**Pull request by johnzheng1975**
Envoy rate limit is a important feature, after old istio rate limit (based on adapt) is obsolete.

However, current example has no such deploy file: https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/
I created a PR in istio.io: https://github.com/istio/istio.io/pull/9140
bianpengyuan suggest I create deploy file here, and refer by  istio.io.

 Note that this had been test pass by me.

[ ] Configuration Infrastructure
[x] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by bianpengyuan** [comment]
@johnzheng1975 thanks a lot!

Could you please:
1) remove the duplicated part in this file https://github.com/istio/istio/blob/master/tests/integration/telemetry/policy/testdata/ratelimitservice.yaml and rename it to rate-limit-configmap.yaml.
2) Update the logic here: https://github.com/istio/istio/blob/5f889798ed3ee28ec80cc7c08afd2e42a720fe6b/tests/integration/telemetry/policy/envoy_ratelimit_test.go#L130 to apply the new sample file that you added, and the new `rate-limit-configmap.yaml` created at 1).

**Comment by johnzheng1975** [comment]
Thanks a lot for your comments:
Sorry I did not get it well.

> 1. remove the duplicated part in this file https://github.com/istio/istio/blob/master/tests/integration/telemetry/policy/testdata/ratelimitservice.yaml and rename it to rate-limit-configmap.yaml.
John:  Rename samples/ratelimit/ratelimit.yaml? Or rename  https://github.com/istio/istio/blob/master/tests/integration/telemetry/policy/testdata/ratelimitservice.yaml  to rate-limit-configmap.yaml?
Note that my new file is copy from https://github.com/istio/istio/blob/master/tests/integration/telemetry/policy/testdata/ratelimitservice.yaml exactly, not sure how to remove the duplicate part. 

Thanks a lot for your patient answer. I want to do a try for istio repo contribution.

**Comment by johnzheng1975** [comment]
BTW, I wrote a example for ratelimit in my own repo, test pass.
I can put all of them in if you think so.
https://github.com/johnzheng1975/istio1.9-ratelimit-example/blob/main/README.md
But this is another topic, Just FYI

**Comment by bianpengyuan** [comment]
> Also, the istio.io doc has no redis/ratelimit deploy/svc, but not lack rate-limit-configmap.yaml, should we put it here?

Yeah, I meant keep only the deployment and service (as what you have in this PR) under sample folder, trim the duplicated parts in the testdata file, but keep the configmap under the testdata folder (which is the only diff with the sample you are adding). In the test, apply `sample/rate-limit-svc.yaml` and `testdata/rate-limit-configmap.yaml`.

**Comment by johnzheng1975** [comment]
Done for your recommendation. Please help to review. Thanks.

File: tests/integration/telemetry/policy/envoy_ratelimit_test.go [hunk 1]
```diff
@@ -127,7 +127,19 @@ func testSetup(ctx resource.Context) (err error) {
...
+	}
+
+	err = ctx.Config().ApplyYAML(ratelimitNs.Name(),
+		string(yamlContentCM),
+	)
+	if err != nil {
+		return
+	}
+
+	yamlContent, err := ioutil.ReadFile("../../../../samples/ratelimit/ratelimit.yaml")
```
Comment (bianpengyuan) [inline tests/integration/telemetry/policy/envoy_ratelimit_test.go:142]: nit: `filepath.Join(env.IstioSrc, "samples/ratelimit/ratelimit.yaml")`

**Review by bianpengyuan — COMMENTED** [review]
Looks good! lint complains about rate limit port naming. Could you add protocol prefix to them?

File: tests/integration/telemetry/policy/envoy_ratelimit_test.go (hunk 1 above)
Comment (johnzheng1975) [reply tests/integration/telemetry/policy/envoy_ratelimit_test.go]: Done

**Comment by johnzheng1975** [comment]
> Looks good! lint complains about rate limit service port naming. Could you add protocol prefix to them?

Done.
```
spec:
   ports:
-  - name: "8080"
+  - name: http
     port: 8080
     targetPort: 8080
     protocol: TCP
-  - name: "8081"
+  - name: grpc
     port: 8081
     targetPort: 8081
     protocol: TCP
-  - name: "6070"
+  - name: debug
     port: 6070
     targetPort: 6070
     protocol: TCP
```

**Comment by bianpengyuan** [comment]
The requirement is to add protocol prefix. so instead of `debug` it should be `http-debug`

**Comment by johnzheng1975** [comment]
> The requirement is to add protocol prefix. so instead of `debug` it should be `http-debug`

Done. Thanks for your time.

**Comment by ericvn** [comment]
You can click on `Details` after the lint test and then click on the raw build log to see the failures. Complaint is:
```
tests/integration/telemetry/policy/envoy_ratelimit_test.go:20: File is not `gofumpt`-ed (gofumpt)
	"io/ioutil"
tests/integration/telemetry/policy/envoy_ratelimit_test.go:23: File is not `gofumpt`-ed (gofumpt)
	"path/filepath"
tests/integration/telemetry/policy/envoy_ratelimit_test.go:26: File is not `gofumpt`-ed (gofumpt)
	"istio.io/istio/pkg/test/echo/common/response"
tests/integration/telemetry/policy/envoy_ratelimit_test.go:38: File is not `gofumpt`-ed (gofumpt)
	"istio.io/istio/pkg/test/env"
```

**Comment by johnzheng1975** [comment]
make gen

**Review by bianpengyuan — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Pull request merged**